### PR TITLE
HandShake and Entanglement and Incompatible Measurements Running Through It

### PR DESCRIPTION
**Pull request by LSchatzki**
Hello Everybody,

As I said earlier, I put together a draft of handshake and adjusted entanglement and incompatible measurement and run through it. Handshake takes in a communicator as a member variable because the tests shouldn't know what type of communicator to make. I also changed entanglement to be a collection of functions. Both tests have a function called run_test() now that runs the test and outputs pass/fail and the test value. These don't need to be integrated immediately, just want to go over this format and decide if we like it.

### CODE REVIEW

File: device_independent_test/entanglement.py [hunk 1]
```diff
@@ -1,103 +1,69 @@
...
-    def measure_in_ZV(self):
-        circuit = self.create_bell_state()
-        circuit.s(1)
-        circuit.h(1)
-        circuit.tdg(1)
-        circuit.h(1)
-        circuit.measure_all()
-        return circuit
+def measure_in_ZV():
+         circuit = QuantumCircuit(2)
```
Comment (m-ludi) [inline device_independent_test/entanglement.py:43]: indentations

File: device_independent_test/handshake.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+from . import incompatible_measurement
+from . import new_entanglement
+
+class HandShake():
+
+    def __init__(self,communicator: quantum_communicator.QuantumDispatcher):
+        self.dispatcher = communicator
+
+    def dimensionality(self,tolerance,shots):
+        pass
```
Comment (m-ludi) [inline device_independent_test/handshake.py:11]: NP: new lines between the methods for readability and similar below

File: device_independent_test/handshake.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
+from . import quantum_communicator
+from . import incompatible_measurement
+from . import new_entanglement
+
+class HandShake():
```
Comment (m-ludi) [inline device_independent_test/handshake.py:5]: maybe some documentation describing what this does

File: device_independent_test/handshake.py [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+        result = new_entanglement.run_test(self.dispatcher,tolerance,shots)
+        if result[0]:
+            print("Passed Entanglement with violation: ", result[1])
+        else:
+            print("Failed Entanglement with violation: ", result[1])
+        return result
+    def test_all(self,tolerance,shots):
+        pass
+
+    def test(self,name,tolerance,shots):
```
Comment (m-ludi) [inline device_independent_test/handshake.py:29]: I think this method is not really necessary. We can rename `entanglement` to `test_entanglement`, and the user can use them directly. I don't think it's a good practice to use `getattr` if possible.

File: device_independent_test/handshake.py [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+        try:
+            getattr(self,name)(tolerance,shots)
+        except:
+            print("Test not found.")
+            self.print_tests()
+
+    def print_tests(self):
```
Comment (m-ludi) [inline device_independent_test/handshake.py:36]: This can also go I think.

File: device_independent_test/handshake.py (hunk 4 above)
Comment (LSchatzki) [reply device_independent_test/handshake.py:29]: Removed

File: device_independent_test/handshake.py (hunk 5 above)
Comment (LSchatzki) [reply device_independent_test/handshake.py:36]: It's not really a big deal but I kinda like leaving it in.

File: device_independent_test/entanglement.py [hunk 6]
```diff
@@ -1,103 +1,69 @@
...
 from qiskit import IBMQ, QuantumCircuit, execute
-from qiskit.tools.monitor import job_monitor
-from qiskit.providers.ibmq.managed import IBMQJobManager
 
-class EntanglementTest:
-    @staticmethod
-    def create_bell_state():
-        """Creates two qubit bell state
+def create_bell_state():
+      """Creates two qubit bell state
```
Comment (m-ludi) [inline device_independent_test/entanglement.py:5]: indentation tab size should be 4 to be consistent. I think other parts of this file should be reindented.

**Pull request merged**